Defines in options YAML file not taken in account with gcov

We have many projects using Ceedling for the unit tests and working well, but we just found a configuration that seems not to work, when using both gcov and extra YAML option files.

Those configurations work fine:
- One YAML file (project.yml) and gcov
- One main YAML file (project.yml) and multiple _options_ YAML files used with _ceedling options:some_file test:some_test_

However, it seems that when we use multiple _options_ YAML files while trying to obtain the coverage with gcov, **the defines placed in the _options_ file are not passed to the compiler**.

Here is a **minimal project** reproducing the issue:
[project.zip](https://github.com/ThrowTheSwitch/Ceedling/files/2358030/project.zip)

It has a source file (foo.c) which cannot build unless **FOOBAR** is defined. There are two YAML files, the main one (project.yml) and an option one (yml_options/foobar.yml) which defines FOOBAR.

This does not work, as expected since **FOOBAR** is not defined:
```
ceedling test:test_foo
=> src/foo.c:4:2: error: #error "FOOBAR NOT DEFINED"
 #error "FOOBAR NOT DEFINED"
```

This works fine since **FOOBAR** is defined : 
```
ceedling options:foobar test:test_foo
```

And here's the problem, this build fails:
```
ceedling options:foobar gcov:test_foo
=> src/foo.c:4:2: error: #error "FOOBAR NOT DEFINED"
 #error "FOOBAR NOT DEFINED"
```

So the question is, how to generate gcov reports while running the tests with an  _options_ YAML file ? Is this a bug/missing feature or is there something wrong with our configuration?

Thank you for any help you can provide and for the amazing work on Ceedling&co!

**System**
```
Windows 7 64 bit
Ceedling:: 0.28.3
CException:: 1.3.1.18
CMock:: 2.4.6.217
Unity:: 2.4.3.122
```

--- a/plugins/gcov/config/defaults.yml
+++ b/plugins/gcov/config/defaults.yml
@@ -11,6 +11,7 @@
       - -I"$": COLLECTION_PATHS_TEST_TOOLCHAIN_INCLUDE
       - -D$: COLLECTION_DEFINES_TEST_AND_VENDOR
       - -DGCOV_COMPILER
+      - -DCODE_COVERAGE
       - -c "${1}"
       - -o "${2}"
   :gcov_linker:

--- a/plugins/gcov/lib/gcov.rb
+++ b/plugins/gcov/lib/gcov.rb
@@ -14,7 +14,6 @@ class Gcov < Plugin
       project_test_results_path: GCOV_RESULTS_PATH,
       project_test_dependencies_path: GCOV_DEPENDENCIES_PATH,
       defines_test: DEFINES_TEST + ['CODE_COVERAGE'],
-      collection_defines_test_and_vendor: COLLECTION_DEFINES_TEST_AND_VENDOR + ['CODE_COVERAGE'],
       gcov_html_report_filter: GCOV_FILTER_EXPR
     }
 

